ksp_plugin_adapter/ksp_plugin_adapter.cs

```diff
@@ -223,29 +221,43 @@ private enum UnityLayers {
   private Dictionary<uint, QP> part_id_to_degrees_of_freedom_ =
       new Dictionary<uint, QP>();
 
-  // The first apocalyptic error message.
+  // UI for the apocalypse notification.
   [KSPField(isPersistant = true)]
   private String revelation_ = "";
-  // Whether we have encountered an apocalypse already.
   [KSPField(isPersistant = true)]
   private bool is_post_apocalyptic_ = false;
   [KSPField(isPersistant = true)]
-  private int apocalypse_window_x_ = UnityEngine.Screen.width / 2;
+  private int apocalypse_dialog_x_ = Dialog.XCentre;
   [KSPField(isPersistant = true)]
-  private int apocalypse_window_y_ = UnityEngine.Screen.height / 3;
-  private UnityEngine.Rect apocalypse_window_rectangle_;
+  private int apocalypse_dialog_y_ = Dialog.YCentre;
+  private Dialog apocalypse_dialog_ = new Dialog();
+
+  // UI for the bad installation notification.
+  [KSPField(isPersistant = true)]
+  private String bad_installation_message_ = "";
+  [KSPField(isPersistant = true)]
+  private bool is_bad_installation_ = false;
```

Comment:
Why would you want to persist this bit?
If the user is missing the redistributable framework, sees the message, installs it, restarts KSP (now with a good Principia installation), and reloads the save, the now-outdated error message will be shown; the save will forever be doomed, whereas there never was a problem with the save itself.

Reply:
Changed to not persist the message and the bit.